src/hooks/attachments/useAttachment.ts

```diff
@@ -222,10 +224,20 @@ export default function useAttachment(
     }
   }, [value])
 
+  const canDownload = React.useMemo(() => {
+    const isDownloadableAttachment = (attachment: Attachment) =>
+      !attachment.isPrivate || isLoggedIn
```

Comment:
```suggestion
      !!attachment.type || !attachment.isPrivate || isLoggedIn
```

Should be allowing a private file to be downloaded if it has not been uploaded yet.

Reply:
IMO, these types of memos are better suited to this format (favouring maintainability and readability over less code)

```js
if (firstPredicate) {
  return true
}

if (secondPredicate) {
  return true
}

if (thirdPredicate) {
  return true
}

return lastPredicate
```

Reply:
Are you sure? I thought we wanted to deny downloading if it was private and you weren't logged in? Even it it hasnt uploaded yet. Doesn't make much sense to be inconsistent about that, even though there is a technical difference, user won't perceive it that way.

Reply:
Agreed